### PR TITLE
[Overlay] fix wrong mountpoint in certain conditions

### LINKED ISSUE #3043: Wrong overlay mount-point when rendering in another web page using React.createPortal
#### Environment

- __Package version(s)__: 3.7.0

#### Steps to reproduce

1. open the example https://codesandbox.io/s/30kyp019vq
1. click `open a portal` button

#### Actual behavior

A tooltip is rendered in the original webpage

image

#### Expected behavior

The tooltip should be rendered on the opened window

image

#### Possible solution

set `usePortal` to false can avoid this problem, but there are some behavior differences on enable `usePortal` or not

### PR DESCRIPTION
**Pull request by KochiyaOcean**
#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [ ] Include tests
- [x] Update documentation

#### Changes proposed in this pull request:

Added `document` prop to `<Portal>` component, which allows to render elements to another DOM.

Changed some behavior of `<Overlay>` component.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @KochiyaOcean! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by giladgray** [comment]
this is not a use case i'm keen to support but it should actually be handled by the proposal in #3033. please read that issue and try again.

File: packages/core/src/components/portal/portal.tsx [hunk 1]
```diff
@@ -96,7 +101,7 @@ export class Portal extends React.Component<IPortalProps, IPortalState> {
     }
 
     private createContainerElement() {
-        const container = document.createElement("div");
+        const container = (this.props.document || document).createElement("div");
```
Comment (giladgray) [inline packages/core/src/components/portal/portal.tsx]: use defaultProps for this.

File: packages/core/src/components/portal/portal.tsx [hunk 2]
```diff
@@ -21,6 +21,11 @@ export interface IPortalProps extends IProps {
      * Callback invoked when the children of this `Portal` have been added to the DOM.
      */
     onChildrenMount?: () => void;
+
+    /**
+     * The document object of DOM that children will be mounted to. Default value is `window.document`.
+     */
+    document?: Document;
```
Comment (giladgray) [inline packages/core/src/components/portal/portal.tsx]: prefer `container` per #3033 and the type is more permissive, like any Element.

**Comment by KochiyaOcean** [comment]
@giladgray all requested changes have done.

**Pull request closed without merging**